corehq/apps/cloudcare/static/cloudcare/js/formplayer/menus/api.js

```diff
@@ -53,23 +53,29 @@ hqDefine("cloudcare/js/formplayer/menus/api", function () {
                                 menus.fetch($.extend(true, {}, options, { data: newOptionsData }));
                             }, gettext('Waiting for server progress'));
                         } else if (_.has(response, 'exception')) {
-                            FormplayerFrontend.trigger('clearProgress');
-                            FormplayerFrontend.trigger(
-                                'showError',
-                                response.exception,
-                                response.type === 'html'
-                            );
-
-                            var currentUrl = FormplayerFrontend.getCurrentRoute();
-                            if (FormplayerFrontend.lastError === currentUrl) {
-                                FormplayerFrontend.lastError = null;
-                                FormplayerFrontend.trigger('navigateHome');
+                            if (params.clickedIcon &&
+                                response.statusCode === 404) {
+                                parsedMenus.removeCaseRow = true;
+                                FormplayerFrontend.trigger('clearProgress');
+                                defer.resolve(parsedMenus);
                             } else {
-                                FormplayerFrontend.lastError = currentUrl;
-                                FormplayerFrontend.trigger('navigation:back');
+                                FormplayerFrontend.trigger('clearProgress');
```

Comment:
this line is in both if and else so can it be outside of the conditional?

Reply:
fixed here https://github.com/dimagi/commcare-hq/pull/33802/commits/db4127aad84631b346d5cbba56520ee96b8d724a